packages/fast-navigation-generator-react/src/navigation/navigation.tsx

```diff
@@ -0,0 +1,392 @@
+import * as React from "react";
+import { KeyCodes } from "@microsoft/fast-web-utilities";
+import Foundation, {
+    FoundationProps,
+    HandledProps,
+} from "@microsoft/fast-components-foundation-react";
+import { get } from "lodash-es";
+import { canUseDOM } from "exenv-es6";
+import {
+    NavigationHandledProps,
+    NavigationProps,
+    NavigationState,
+    NavigationUnhandledProps,
+    TreeNavigation,
+} from "./navigation.props";
+import { getNavigationFromData } from "./navigation.utilities";
+
+export default class Navigation extends Foundation<
+    NavigationHandledProps,
+    NavigationUnhandledProps,
+    NavigationState
+> {
+    public static displayName: string = "Navigation";
+
+    protected handledProps: HandledProps<NavigationHandledProps> = {
+        schema: void 0,
+        data: void 0,
+        childOptions: void 0,
+        onLocationUpdate: void 0,
+        managedClasses: void 0,
+    };
+
+    private rootElement: React.RefObject<HTMLDivElement>;
+
+    constructor(props: NavigationProps) {
+        super(props);
+
+        this.state = {
+            navigation: getNavigationFromData(
+                this.props.data,
+                this.props.schema,
+                this.props.childOptions
+            ),
+            openItems: [],
+            activeItem: null,
+        };
+
+        this.rootElement = React.createRef();
+    }
+
+    public render(): React.ReactNode {
+        return (
+            <div
+                {...this.unhandledProps()}
+                ref={this.rootElement}
+                role={"tree"}
+                className={this.props.managedClasses.navigation}
+            >
+                {this.renderTreeItems(this.state.navigation, 1)}
+            </div>
+        );
+    }
+
+    /**
+     * Renders the tree item containing element
+     */
+    private renderTreeItemContainer(
+        navigation: TreeNavigation[],
+        level: number
+    ): React.ReactNode {
+        return (
+            <div className={this.props.managedClasses.navigation_itemList}>
```

Comment:
You'll need `role={"group"}` on this div since it is the container of child tree-items